src/util/BlankNodeManager.h

```diff
@@ -37,23 +48,52 @@ class BlankNodeManager {
       (ValueId::maxIndex - minIndex_ + 1) / blockSize_;
 
  private:
-  // Int Generator yielding random block indices.
-  SlowRandomIntGenerator<uint64_t> randBlockIndex_;
+  // A necessary forward declaration.
+  struct Blocks;
 
-  // Tracks blocks currently used by instances of `LocalBlankNodeManager`.
-  Synchronized<HashSet<uint64_t>> usedBlocksSet_;
+  // We identify sets of blocks by UUIDs, and therefore need to hash UUIDs`.
+  struct UuidHash {
+    std::size_t operator()(const boost::uuids::uuid& uuid) const {
+      return boost::hash<boost::uuids::uuid>()(uuid);
+    }
+  };
```

Comment:
Can't you just do

```suggestion
  using UuidHash = boost::hash<boost::uuids::uuid>;
```

or use the type directly down below?